src/msg/async/Stack.h

```diff
@@ -310,6 +310,14 @@ class NetworkStack {
   virtual void spawn_worker(unsigned i, std::function<void ()> &&) = 0;
   virtual void join_worker(unsigned i) = 0;
 
+  virtual void handle_pre_fork() override {
```

Comment:
why is `NetworkStack` started before `daemon()`? since the glibc implements `daemon(2)` using `fork()`, it's not safe to do any threading stuff before calling `daemon()`, i think.